source/state_representation/src/space/joint/JointState.cpp

```diff
@@ -91,6 +89,13 @@ JointState& JointState::operator=(const JointState& state) {
   return *this;
 }
 
+void JointState::resize(unsigned int size) {
```

Comment:
There can be some edge cases where resizing without resetting data may be desirable. For example, "slicing" some joints from the end. Since this is a private method I think it's good how it is. Also, mixing behaviors in functions should be avoided anyway.